var/spack/repos/builtin/packages/nest-simulator/package.py

```diff
@@ -0,0 +1,171 @@
+    depends_on('pkg-config', type='build')
+
+    extends('python', when='+python')
+
+    # Before 2.12.0 it was an autotools package
+    @when('@:2.10.99')
+    def cmake(self, spec, prefix):
+        pass
+
+    @when('@:2.10.99')
+    def build(self, spec, prefix):
+        pass
+
+    @when('@:2.10.99')
+    def install(self, spec, prefix):
+        configure_args = ["CXXFLAGS=-std=c++03",
+                          "--prefix=" + prefix,
+                          "--with-openmp"]
+        if '+python' in spec:
+            configure_args.append("--with-python")
+        else:
+            configure_args.append("--without-python")
+        if '+mpi' in spec:
+            configure_args.append("--with-mpi")
+        else:
+            configure_args.append("--without-mpi")
+        if '+optimize' in spec:
+            configure_args.append("--with-optimize")
+        else:
+            configure_args.append("--without-optimize")
+
+        configure(*configure_args)
+
+        make()
+        make("install")
+
+    def cmake_args(self):
+        args = []
+
+        if '+mpi' in self.spec:
+            args.append('-Dwith-mpi=ON')
+        else:
+            args.append('-Dwith-mpi=OFF')
+
+        if '+python':
+            version = self.spec['python'].version[0]
+            args.append('-Dwith-python={0}'.format(version))
+            args.append('-Dcythonize-pynest=' + self.spec['py-cython'].prefix)
+        else:
+            args.append('-Dwith-python=OFF')
+            args.append('-Dcythonize-pynest=OFF')
+
+        if '+optimize' in self.spec:
+            args.append('-Dwith-optimize=ON')
+        else:
+            args.append('-Dwith-optimize=OFF')
+
+        if '+gsl' in self.spec:
+            cmake_options.append('-Dwith-gsl=' + self.spec['gsl'].prefix)
+        else:
+            cmake_options.append('-Dwith-gsl=OFF')
+
+        if '+openmp' in self.spec:
+            cmake_options.append('-Dwith-openmp=ON')
+        else:
+            cmake_options.append('-Dwith-openmp=OFF')
+
+        if '+shared' in self.spec:
+            cmake_options.append('-Dstatic-libraries=OFF')
+        else:
+            cmake_options.append('-Dstatic-libraries=ON')
+
+        return args
+
+    @when('@:2.14.0+modules')
+    @run_after('install')
+    def install_headers(self):
+        # copy source files to installation folder for older versions
+        # (these are needed for modules to build against)
+        # see https://github.com/nest/nest-simulator/pull/844
+        path_headers = join_path(prefix, "include", "nest")
+
+        mkdirp(path_headers)
+
+        for suffix in ["h", "hpp"]:
+            for f in find_headers('*.{0}'.format(suffix),
+                                  self.stage.source_path, recursive=True):
+                install(f, path_headers)
+
+    def setup_environment(self, spack_env, run_env):
+        run_env.set("NEST_INSTALL_DIR", self.spec.prefix)
+        run_env.set('NEST_EXE', self.prefix.bin.nest)
```

Comment:
these env variables were for our convenience . I would remove this from upstream.

Reply:
`NEST_INSTALL_DIR` seems to be required for building nest-modules -> only removing `NEST_EXE`